read_terraform_state_policy.tf

```diff
@@ -0,0 +1,39 @@
+# ------------------------------------------------------------------------------
+# Create the IAM policy that allows read-only access to the specified
+# Terraform state in the S3 bucket where Terraform remote state is stored.
+# This is useful for cases when read-only access to a particular state is
+# needed, but read-only access to other Terraform states in the bucket is not.
+# ------------------------------------------------------------------------------
+
+# IAM policy document that allows read-only access to the specified state
+# in the Terraform state bucket.
+data "aws_iam_policy_document" "read_terraform_state" {
+  statement {
+    actions = [
+      "s3:ListBucket",
+    ]
+    resources = [
+      "arn:aws:s3:::${var.terraform_state_bucket_name}",
+    ]
+  }
+
+  statement {
+    actions = [
+      "s3:GetObject",
+    ]
+    resources = concat(
+      # If the "default" workspace (or "*" for all workspaces) is specified,
+      # include the path where Terraform stores default workspace state.
+      contains(["*", "default"], var.terraform_workspace) ? ["arn:aws:s3:::${var.terraform_state_bucket_name}/${var.terraform_state_path}"] : [],
+      # For non-"default" workspaces, include the correct bucket path.
+      var.terraform_workspace != "default" ? ["arn:aws:s3:::${var.terraform_state_bucket_name}/env:/${var.terraform_workspace}/${var.terraform_state_path}"] : [],
```

Comment:
This all comes from the backend configuration?  Just making sure it's not fragile.
https://www.terraform.io/docs/language/settings/backends/s3.html#key

Reply:
It's a bit fragile- if Terraform were to change the path(s) where it stores state objects in S3, this would break.  However I'm not sure how to get around that.